### PR TITLE
Zoltan2: Add SARMA library as TPL

### PR DESCRIPTION
**Pull request by mbenlioglu**
## Motivation
Adds an interface to call [SARMA](https://github.com/GT-TDAlab/SARMA) partitioning library

### CODE REVIEW

**Comment by cgcgcg** [comment]
@trilinos/zoltan2

File: packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgSarma.hpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        pe = sparams.getEntryPtr("p"); // row_cut
+        if (pe)
+            config.row_parts = pe->getValue(&config.row_parts); //col_cut
+        pe = sparams.getEntryPtr("q");
+        if (pe)
+            config.col_parts = pe->getValue(&config.col_parts); // max_load
+        pe = sparams.getEntryPtr("z");
+        if (pe)
+            config.z = pe->getValue(&config.z);
+        pe = sparams.getEntryPtr("sparsify");
```
[the remark this answers is not in the document]
Comment (mbenlioglu) [reply packages/zoltan2/core/src/algorithms/partition/Zoltan2_AlgSarma.hpp]: Just updated

**Review by egboman — COMMENTED** [review]
As discussed, please add a test in Zoltan2 that tests SARMA. The test can be very simple. Please put it in Trilinos/packages/zoltan2/test/core/partition. There are a couple small matrices there already that can be used in the tests. The test should of course only be run if Trilinos/Zoltan2 was configured/built with SARMA. 

Otherwise, looks good. Thanks.

**Comment by egboman** [comment]
@mbenlioglu showed me that there is now a working test for SARMA as TPL. It runs in serial or with MPI but only one process.

**Review by egboman — DISMISSED** [review]
The missing test was added so I now approve the PR. Thanks, Mucahid.

**Review by srajama1 — APPROVED** [review]
@egboman has approved it, but github is complaining 1 pending reviewer, so I will approve as well so tests can start. Thanks @mbenlioglu !

**Review by egboman — APPROVED** [review]
Minor fix for Sarma header, OK.

**Review by srajama1 — APPROVED** [review]
Thanks @mbenlioglu !

**Pull request merged**